tests/parametric/test_headers_b3.py

```diff
@@ -102,8 +112,140 @@ def test_headers_b3multi_propagate_valid(self, test_agent, test_library):
 
     @enable_b3()
     @missing_feature(context.library == "cpp", reason="format of DD_TRACE_PROPAGATION_STYLE_EXTRACT not supported")
-    @missing_feature(context.library == "ruby", reason="Ruby doesn't support case-insensitive distributed headers")
-    def test_headers_b3multi_propagate_invalid(self, test_agent, test_library):
+    @missing_feature(context.library == "ruby", reason="format of DD_TRACE_PROPAGATION_STYLE_EXTRACT not supported")
+    def test_headers_b3_propagate_invalid(self, test_agent, test_library):
+        """Ensure that invalid b3 distributed tracing headers are not extracted
+        and the new span context is injected properly.
+        """
+        with test_library:
+            headers = make_single_request_and_get_inject_headers(test_library, [["b3", "0-0-1"]])
+
+        span = get_span(test_agent)
+        assert span.get("trace_id") != 0
+        assert span.get("span_id") != 0
+
+        b3Arr = headers["b3"].split("-")
+        b3_trace_id = b3Arr[0]
+        b3_span_id = b3Arr[1]
+        b3_sampling = b3Arr[2]
+
+        assert len(b3_trace_id) == 16 or len(b3_trace_id) == 32
+        assert int(b3_trace_id, base=16) == span.get("trace_id")
+        assert int(b3_span_id, base=16) == span.get("span_id") and len(b3_span_id) == 16
+        assert b3_sampling == "1" if span["metrics"].get(SAMPLING_PRIORITY_KEY) > 0 else "0"
+        assert span["meta"].get(ORIGIN) is None
+
+    @enable_migrated_b3()
+    @missing_feature(context.library == "cpp", reason="format of DD_TRACE_PROPAGATION_STYLE_EXTRACT not supported")
+    @missing_feature(context.library == "dotnet", reason="Need to remove b3=b3multi alias")
+    @missing_feature(context.library == "golang", reason="Need to remove b3=b3multi alias")
+    @missing_feature(context.library == "java", reason="Need to remove b3=b3multi alias")
+    @missing_feature(context.library == "nodejs", reason="Need to remove b3=b3multi alias")
+    @missing_feature(context.library == "php", reason="Need to remove b3=b3multi alias")
+    @missing_feature(context.library == "python", reason="Need to remove b3=b3multi alias")
+    @missing_feature(context.library == "python_http", reason="Need to remove b3=b3multi alias")
+    def  test_headers_b3_migrated_extract_valid(self, test_agent, test_library):
```

Comment:
The remaining tests here are duplicates of the ones above except they use the new configuration "b3" instead of "b3 single header"